mcs/class/System/Internal.Cryptography/X509Pal.cs

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) 2018 Microsoft
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Internal.Cryptography.Pal
+{
+	// Temporary until real X509Pal from CoreFX could be used.
+	internal sealed partial class X509Pal
+	{
+		public static ManagedX509Pal Instance = new ManagedX509Pal ();
+
+		private X509Pal ()
+		{
+		}
+	
+		public partial class ManagedX509Pal : ManagedX509ExtensionProcessor
```

Comment:
Let's make this an internal top-level class and find a better name for it.  We are currently using names such as for instance `X509ChainImpl`, but I wouldn't mind transitioning to a new "Pal" based naming scheme.

Where does `ManagedX509ExtensionProcessor` come from, is that a new CoreFX type?

Reply:
The X509Pal name is referenced in the CoreFX code, the ManagedX509Pal class is just temporary one to make this small piece of code compile without the full platform specific backends.

ManagedX509ExtensionProcessor comes from CoreFX and implements all the X509Extension code on all non-Windows platforms (but works on Windows too).